docs/advanced-features/automatic-static-optimization.md

```diff
@@ -20,6 +20,8 @@ If the above is not the case, Next.js will **statically optimize** your page aut
 
 During prerendering, the router's `query` object will be empty since we do not have `query` information to provide during this phase. After hydration, Next.js will trigger an update to your application to provide the route parameters in the `query` object.
 
+However, with `Rewrites` in your code, a re-render is triggered to provide the parsed parameters. This is because [rewrites](/docs/api-reference/next.config.js/rewrites) can have parameters that we need to provide in the `query` object.
```

Comment:
I would mention `router.isReady` as well and link to https://nextjs.org/docs/api-reference/next/router#router-object for those that need to know when it is safe to start using `query`.

I would also backlink from `isReady` https://nextjs.org/docs/api-reference/next/router#router-object to this page so the purpose of `isReady` is more clear.

Reply:
added in https://github.com/vercel/next.js/pull/34049/commits/c7e50643d4a268852fd719dc9993fff2f7babe90